back/engines/commercial/insights/app/services/insights/category_suggestions_service.rb

```diff
@@ -60,6 +63,15 @@ def classify(inputs, categories)
       create_tasks(tasks_infos)
     end
 
+    # @param [Enumerable<Insights::ZeroshotClassificationTask>] tasks
+    # @return [Array<Insights::ZeroshotClassificationTask>] Deleted frozen tasks
+    def cancel(tasks)
+      # We need to use identifiers assigned by the NLP service, not the identifier of the record.
+      nlp_client.cancel_tasks(tasks.pluck(:task_id))
```

Comment:
Shouldn't we check the response code here? And not to delete the tasks if it's not successful. I suppose the service can be down or something like that.

Reply:
I hesitated a bit there, but in the end, I opted for that solution to keep things simple, especially at the level of the front-end. I think it's ok-ish because if the request fails, the NLP service will just run the tasks but the back-end will just happily discard the result. But maybe it would be worth to send something to sentry when the request fails?

Reply:
> the NLP service will just run the tasks

Indeed, it's not a big deal.

> But maybe it would be worth to send something to sentry when the request fails?

I think no need if it's already sent in NLP.

Reply:
I don't think it does 😬

Reply:
But we have some errors there https://sentry.hq.citizenlab.co/organizations/citizenlab/issues/?project=10
Are they sent by our custom code? Not with integration?

Reply:
Indeed but it does not do consistently. Also, in that case, if it returns a 400 it would not report anything to sentry (since the error would be on the back-end side).

Reply:
I see. Then it really makes sense to report it to Sentry. See back/lib/error_reporter.rb

Reply:
Was about to ask 🙂